code/modules/xenoarcheaology/effect_master.dm

```diff
@@ -0,0 +1,379 @@
+
+/*
+ * Here there be dragons.
+ */
+
+/atom/var/datum/artifact_master/artifact_master
+
+/atom/Initialize()
+	. = ..()
+
+	if(ispath(artifact_master))
+		artifact_master = new artifact_master(src)
+
+/atom/Destroy()
+	if(artifact_master)
+		QDEL_NULL(artifact_master)
+
+	. = ..()
+
+/atom/attack_hand(var/mob/user as mob)
+	. = ..()
+
+	if(artifact_master)
+		artifact_master.on_attack_hand(user)
+
+/atom/attackby(obj/item/weapon/W as obj, mob/living/user as mob)
+	. = ..()
+
+	if(artifact_master)
+		artifact_master.on_attackby(W, user)
+
+/atom/Bumped(M as mob|obj)
```

Comment:
I don't believe `as mob|obj` does anything in this context. The useful places for `as blah` are inputs, verb args, and for loops.